chunk/chunk_store.go

```diff
@@ -417,14 +417,28 @@ func (c *AWSStore) Get(ctx context.Context, from, through model.Time, matchers .
 	}
 
 	// TODO push ctx all the way through, so we can do cancellation (eventually!)
-	missing, err := c.lookupChunks(userID, from, through, matchers)
+	chunks, err := c.lookupChunks(userID, from, through, matchers)
 	if err != nil {
 		return nil, err
 	}
 
-	queryChunks.Observe(float64(len(missing)))
+	filtered := make([]Chunk, 0, len(chunks))
+	for _, chunk := range chunks {
+		_, chunkFrom, chunkThrough, err := parseChunkID(chunk.ID)
```

Comment:
I actually like `from` & `through` - `end` tends to be a reserved word in many languages, and `through` implies inclusivity.